internal/pkg/skuba/replica/replica.go

```diff
@@ -0,0 +1,264 @@
+												"values": [
+													"%s"
+												]
+											}
+										]
+									},
+									"topologyKey": "kubernetes.io/hostname"
+								}
+							]
+						}
+					}
+				}
+			}
+		}
+	}`
+	patchAffinityPreferred = `{
+		"spec": {
+			"template": {
+				"spec": {
+					"affinity": {
+						"podAntiAffinity": {
+							"preferredDuringSchedulingIgnoredDuringExecution": [
+								{
+									"weight": 100,
+									"podAffinityTerm": {
+										"labelSelector": {
+											"matchExpressions": [
+												{
+													"key": "app",
+													"operator": "In",
+													"values": [
+														"%s"
+													]
+												}
+											]
+										},
+										"topologyKey": "kubernetes.io/hostname"
+									}
+								}
+							]
+						}
+					}
+				}
+			}
+		}
+	}`
+	patchReplicas = `{
+		"spec": {
+			"replicas": %v
+		}
+	}`
+)
+
+// Helper provide methods for replica update of deployment
+// resource that has highAvailabilitylabel.
+type Helper struct {
+	Client            clientset.Interface
+	ClusterSize       int
+	SelectDeployments appsv1.DeploymentList
+	Deployment        appsv1.Deployment
+	ReplicaSize       int
+	MinSize           int
+	isUpdated         bool
+}
+
+// NewHelper creates a helper to update deployment replicas.
+func NewHelper(client clientset.Interface) (*Helper, error) {
+	fromSelectors, err := client.AppsV1().Deployments(metav1.NamespaceSystem).List(
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=true", highAvailabilitylabel),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := kubernetes.GetAllNodes(client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Helper{
+		Client:            client,
+		ClusterSize:       len(node.Items),
+		MinSize:           minSize,
+		SelectDeployments: *fromSelectors,
+	}, nil
+}
+
+func (r *Helper) replaceAffinity(remove string, create string) error {
+	// doRemove will remove only when affinity path exists
+	doRemove := false
+	affinity := r.Deployment.Spec.Template.Spec.Affinity
+	if affinity != nil && affinity.PodAntiAffinity != nil {
+		preferredList := len(r.Deployment.Spec.Template.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution)
+		requiredList := len(r.Deployment.Spec.Template.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution)
+		switch {
+		case remove == "preferred":
+			if preferredList != 0 {
+				doRemove = true
+				break
+			}
+			if requiredList != 0 {
+				return nil
+			}
+		case remove == "required":
+			if requiredList != 0 {
+				doRemove = true
+				break
+			}
+			if preferredList != 0 {
+				return nil
+			}
+		}
+	}
+	if doRemove {
+		if err := r.removeAffinity(); err != nil {
+			return err
+		}
+	}
+
+	affinityJSON := fmt.Sprintf(patchAffinityPreferred, r.Deployment.ObjectMeta.Name)
+	if create == "required" {
+		affinityJSON = fmt.Sprintf(patchAffinityRequired, r.Deployment.ObjectMeta.Name)
+	}
```

Comment:
nit: why not use if...else... or switch...case...

Reply:
sure, why not. changed.